go/vt/vtgate/planbuilder/jointree.go

```diff
@@ -92,15 +97,50 @@ type (
 		// The following two fields are used when routing information_schema queries
 		SysTableTableSchema []evalengine.Expr
 		SysTableTableName   map[string]evalengine.Expr
+
+		// these helps in replacing the argNames with the subquery
+		sqToReplace map[string]*sqlparser.Select
 	}
 
 	derivedTree struct {
 		query *sqlparser.Select
 		inner queryTree
 		alias string
 	}
+
+	subqueryTree struct {
+		subquery *sqlparser.Select
+		outer    queryTree
+		inner    queryTree
+		opcode   engine.PulloutOpcode
+		argName  string
+	}
 )
 
+func (s *subqueryTree) tableID() semantics.TableSet {
+	return s.inner.tableID() | s.outer.tableID()
+}
+
+func (s *subqueryTree) cost() int {
+	panic("implement me")
```

Comment:
need a real implementation for this. I think it is `cost(outer)+cost(inner)`, but I'm not sure

Reply:
Fixed in f83e5cbb98